src/main/java/org/wso2/securevault/secret/SecretManager.java

```diff
@@ -214,6 +216,73 @@ public void init(Properties properties) {
         initialized = true;
     }
 
+    /**
+     * Split the secret annotation from the delimiter provided and decides whether to use the legacy provider
+     * or a novel provider to obtain the secret value of the requested alias.
+     *
+     * @param secretAnnotation String contains the alias, the provider type and the repository type.
+     * @return Plain text value of the required secret.
+     */
+    public String resolveSecret(String secretAnnotation) {
+
+        String[] secretAnnotationStrings = secretAnnotation.split(DELIMITER);
+        if (secretAnnotationStrings.length == 1) {
+            if (legacyProvidersExist) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Getting secrets from legacy provider.");
+                }
+                return getSecret(secretAnnotation);
+            }
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("Getting secrets from a provider other than the legacy providers.");
+        }
+        return resolveSecret(secretAnnotationStrings);
+    }
+
+    /**
+     * Resolve the secret annotation for the secrets coming from repositories belongs to providers listed under
+     * secretProviders property.
+     * Repositories categorized under providers other than legacy providers will be identified based on the secret
+     * annotation. Under that, When there is only a single repository, provider type and the repository type should
+     * be assigned otherwise, annotation itself is going to be used to get the above-said values.
+     *
+     * @param annotation Value retrieve by the resolveSecret as the value to be resolved.
+     * @return If there is a secret , otherwise , alias itself.
+     */
+    private String resolveSecret(String[] annotation) throws SecureVaultException {
+
+        int length = annotation.length;
+
+        switch (length) {
+            case 1:
+                if (providers.isEmpty()) {
+                    log.error("No provider has been configured. Returning the annotation itself.");
+                    return Arrays.toString(annotation);
```

Comment:
Why this is converted to String?

Reply:
This method is getting called by the SecretManagerSecretCallbackHandler and previously it calls the getSecret(string alias) of the SecretManager and it passes the string alias upon unsuccessful retrieval of secrets.
To make sure that the flow is not breaking the annotation is converted to a string

Reply:
What is the use case of this array Arrays.toString(annotation); after returning it as a string